splice/queries/reporting.py

```diff
@@ -3,6 +3,49 @@
 from sqlalchemy.sql import func
 from sqlalchemy.orm import aliased
 from sqlalchemy import case
+from splice.queries.common import tuple_to_dict
+
+
+CROSS_DB_GROUP_BY_FIELDS = [
+    'category',
+    'account_id',
+    'campaign_id',
+    'adgroup_id'
+]
+
+
+def build_subquery_table(env, stats_table, group_by, group_by_arr, tiles, filters=None):
+    """
+    Build a subquery for group_by fields that require groupings in mozsplice_campaigns
+    """
+    grouped_tiles = get_tiles(group_by=group_by, filters=filters)
+
+    if not grouped_tiles:
+        return None
+    for group in grouped_tiles:
+        for tile in group['tile_ids']:
+            tiles.append(tile)
```

Comment:
Mutating directly on the argument is kinda confusing here, unless you do want the side-effect here. Prefer to make a copy. Memory is way cheaper than developer's time :laughing: